app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart

```diff
@@ -26,28 +27,22 @@ class _ActerIconWidgetState extends State<ActerIconWidget> {
 
   @override
   Widget build(BuildContext context) {
-    if (widget.color != null) {
-      color.value = widget.color!;
-    }
-    if (widget.icon != null) {
-      icon.value = widget.icon!;
-    }
+    widget.color.let((p0) => color.value = p0);
+    widget.icon.let((p0) => icon.value = p0);
```

Comment:
this is wrong. this should happen in `initState` and `didUpdateDependencies` (pseudocode):

```
  void _setWidgetValues() {
    widget.color.let((p0) => color.value = p0);
    widget.icon.let((p0) => icon.value = p0);
  }

   @override
   void initState() {
     _setWidgetValues()
   }

   @override
   void didUpdateDependencies() {
     _setWidgetValues()
   }
```

Reply:
I think this is maybe why we sometimes see the color not being properly reflected in the UI despite selecting an icon.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/2233/commits/cbe3ef9e87c69d98ee2744633224b93dbb461c8d